### DISCUSSION TITLE
How to Display Flow Labels in Front of the Flow by Default in BPMN

### DISCUSSION THREAD

**Discussion by BhanuT97** [clientIO/joint#2836] [Q&A]
### Introduction

I have configured a BPMN diagram integrated with a React application, where the graph is generated based on JSON input. However, when the diagram is created, the labels for the flows are rendered behind the flow lines by default.

How can I ensure that the labels are displayed in front of the flow lines by default?
<img width="264" alt="BPMNFlow_SS" src="https://github.com/user-attachments/assets/3e71bae3-e876-43f1-85a8-6c78b6fa133a" />

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
It's not below the link. There is no background `<rect/>` or no `fill` color.

The default `bpmn2.Flow` labels have white background. What exactly do you do?
<img width="228" alt="image" src="https://github.com/user-attachments/assets/0087d104-3287-4cf1-a298-874d0bc28544" />

**Reply by BhanuT97** [reply]
I am generating BPMN diagrams from JSON input in a React application. When the diagram is created, the labels for flows are rendered behind the flow lines by default. However, when I select them, they appear correctly in front. Do you know how I can ensure that labels always render in front of the flow lines by default?

**Reply by kumilingus** [reply]
In the context of a single link, JointJS renders the labels always above the link.

If you want the link labels to be above every single cell use [labelsLayer](https://docs.jointjs.com/api/dia/Paper/#labelslayer) paper option.

You need to provide an example, if above won't help.
